ipq806x: add support for Arris TR4400 v2 / RAC2V1A

Lanchon opened the pull request:
@mans0n @dangowrt 

backport to 22.03 required a single line change:
- `nand` renamed back to `nand_controller`

retested everything and it all works.

all macs are fine and wifi performs well. i'm confident that the nvmem-cells mac/pre-cal change is fine so i squashed the commit.

here is a boot log of this exact branch, including wifi client connection to wan:
https://gist.github.com/Lanchon/f9d0762e85b9430668dc09d30bb6ded7

imho, if both branches should be merged, one should cherrypick the other. i would cherrypick from master since a single line change is required afterwards.

thanks!

Lanchon commented:
> === Checking commit '1bb9428d930c105824d00f71c2722ff28a871ba6'
Author name (Lanchon) need to be your real name 'firstname lastname'
Commit subject line seems ok (ipq806x: add support for Arris TR4400 v2 / RAC2V1A)
Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Lanchon <lanchon@gmail.com>')

someone told me i needed to sign off with my real full name. ugly privacy-wise, but i did it. now the system sees "Author name (Lanchon)". i'm not interested in configuring my real name in my Github account. can this be fixed or will i have to not contribute to openwrt ever?

thanks!

dangowrt commented:
You can submit the patch to our mailing list openwrt-devel@lists.openwrt.org (subscribe first on https://lists.openwrt.org ). Make sure to use git-send-email or git-format-patch and also make sure your full real name appears in both, the `From:` header as well as the `Signed-off-by:` line.

Lanchon commented:
thanks Dan. actually my client was misconfigured #facepalm. i fixed it now, thanks again.

Lanchon committed:
ipq806x: add support for Arris TR4400 v2 / RAC2V1A

Hardware specs:
  SoC: Qualcomm IPQ8065 (dual core Cortex-A15)
  RAM: 512 MB DDR3
  Flash: 256 MB NAND, 32 MB NOR
  WiFi: QCA9983 2.4 GHz, QCA9984 5 GHz
  Switch: QCA8337
  Ethernet: 5x 10/100/1000 Mbit/s
  USB: 1x USB 3.0 Type-A
  Buttons: WPS, Reset
  Power: 12 VDC, 2.5 A

Ethernet ports:
  1x WAN: connected to eth2
  4x LAN: connected via the switch to eth0 and eth1
          (eth0 is disabled in OEM firmware)

MAC addresses (OEM and OpenWrt):
  fw_env @ 0x00  d4:ab:82:??:??:?a  LAN (eth1)
  fw_env @ 0x06  d4:ab:82:??:??:?b  WAN (eth2)
  fw_env @ 0x0c  d4:ab:82:??:??:?c  WLAN 2.4 GHz (ath1)
  fw_env @ 0x12  d4:ab:82:??:??:?d  WLAN 5 GHz (ath0)
  fw_env @ 0x18  d4:ab:82:??:??:?e  OEM usage unknown (eth0 in OpenWrt)

  OID d4:ab:82 is registered to:
  ARRIS Group, Inc., 6450 Sequence Drive, San Diego CA 92121, US

More info:
  https://openwrt.org/inbox/toh/arris/tr4400_v2

IMPORTANT:

This port requires moving the 'fw_env' partition prior to first boot to
consolidate 70% of the usable space in flash into a contiguous partition.
'fw_env' contains factory-programmed MAC addresses, SSIDs, and passwords.
Its contents must be copied to 'rootfs_1' prior to booting via initramfs.
Note that the stock 'fw_env' partition  will be wiped during sysupgrade.

A writable 'stock_fw_env' partition pointing to the old, stock location
is included in the port to help rolling back this change if desired.

Installation:

- Requires serial access and a TFTP server.
- Fully boot stock, press ENTER, type in:
mtd erase /dev/mtd21
dd if=/dev/mtd22 bs=128K count=1 | mtd write - /dev/mtd21
umount /config && ubidetach -m 23 && mtd erase /dev/mtd23
- Reboot and interrupt U-Boot by pressing a key, type in:
set mtdids 'nand0=nand0'
set mtdparts 'mtdparts=nand0:155M@0x6500000(mtd_ubi)'
set bootcmd 'ubi part mtd_ubi && ubi read 0x44000000 kernel && bootm'
env save
- Setup TFTP server serving initramfs image as 'recovery.bin', type in:
set ipaddr 192.168.1.1
set serverip 192.168.1.2
tftpboot recovery.bin && bootm
- Use sysupgrade to install squashfs image.

This port is based on work done by AmadeusGhost <amadeus@jmu.edu.cn>.

Signed-off-by: Rodrigo Balerdi <lanchon@gmail.com>

Lanchon commented:
@mans0n @dangowrt 

final workaround for USB bug, thoroughly tested and committed to this PR:
https://github.com/openwrt/openwrt/blob/76e7213547751907e91dbf78456fb45a81f69901/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-tr4400-v2.dts#L347-L358

this essentially backports the fixes from mainline 5.15 dtsi to our 5.10 dtsi additions, but affecting only this device. they come straight from diffing both dtsi files. when openwrt's dtsi is finally fixed, these workarounds will not cause trouble (but will be superfluous of course).

so now this port only enables a single USB block, which turned out to be usb3_1:
https://github.com/openwrt/openwrt/blob/76e7213547751907e91dbf78456fb45a81f69901/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-tr4400-v2.dts#L360-L362

i don't think anyone else will test this port (tapping the serial port is a horrible ordeal on this device), but i did test it thoroughly. with this last blocking issue ironed out, i hope you can consider merging in time for 22.

thanks!

Lanchon commented:
this PR is put on hold because global dtsi fixes were requested: https://github.com/openwrt/openwrt/pull/9692

Lanchon committed:
ipq806x: backport Arris TR4400 v2 / RAC2V1A

Backport support for Arris TR4400 v2 / RAC2V1A to 22.03 branch.

Signed-off-by: Rodrigo Balerdi <lanchon@gmail.com>

Lanchon commented:
this PR is up to date now, but not tested in its latest incarnation (only trivial changes from previous full test).

this is why it is a DRAFT. i will not test again before master is merged.

thanks!

Lanchon commented:
now up to date and fully tested.
